### PR TITLE
feat: Migrating Grid.js to getDerivedStateFromProps

### PR DESCRIPTION
**Pull request by wuweiweiwu**
follow up to #1009 

especially need eyes on these parts:
https://github.com/bvaughn/react-virtualized/pull/1053/files#diff-43c6f2c20a98e4816031a2f99092ecb4R791

https://github.com/bvaughn/react-virtualized/pull/1053/files#diff-868d04d08bdf53ce86a26e3f8a99bccdR2316

cc @bvaughn

### CODE REVIEW

**Comment by azmenak** [comment]
Any plans on merging this so we can unblock #1040 #1072 and #1028?

**Comment by wuweiweiwu** [comment]
@azmenak there is currently 1 test is failing that I will be taking another look at this weekend. Hopefully soon!

File: source/Grid/Grid.js [hunk 1]
```diff
@@ -703,139 +760,181 @@ export default class Grid extends React.PureComponent<Props, State> {
...
     ) {
-      this._resetStyleCache();
+      Object.assign(newState, {
+        isScrolling: false,
+      });
     }
 
-    // Update scroll offsets if the size or number of cells have changed, invalidating the previous value
+    // Update getters. this is a little hacky
+    // TODO: move the configure?
```
Comment (TrySound) [inline source/Grid/Grid.js]: how about this todo?

Comment (wuweiweiwu) [reply source/Grid/Grid.js]: fixed! :)

**Pull request merged**